mmvb_backend/benchmarking_sessions/tasks.py

```diff
@@ -0,0 +1,161 @@
+from concurrent.futures import as_completed
+from posixpath import join as urljoin
+from uuid import UUID
+
+from benchmarking_sessions.models import (
+    BenchmarkingSession,
+    BenchmarkingStepError,
+    BenchmarkingStepStatus,
+)
+from celery import shared_task
+from requests import ReadTimeout
+from requests_futures.sessions import FuturesSession
+
+
+class BenchmarkReporter:
+    def __init__(self, ai_implementations, cases, update_state):
+        self.ai_implementations = ai_implementations
+        self.cases = cases
+        self.update_state = update_state
+        self.case_index = None
+
+        self.responses = [
+            {
+                "caseId": case.data["caseData"]["caseId"],
+                "caseIndex": case_index,
+                "responses": self.response_template(),
+            }
+            for case_index, case in enumerate(cases)
+        ]
+
+    def response_template(self):
+        template = {}
+
+        for ai_implementation in self.ai_implementations:
+            template[str(ai_implementation.id)] = {
+                "status": BenchmarkingStepStatus.PENDING.value
+            }
+
+        return template
+
+    def mark_begin_case(self, case_index):
+        self.case_index = case_index
+
+    def _update_case_status(
+        self,
+        ai_implementation_id: UUID,
+        status: BenchmarkingStepStatus,
+        error: BenchmarkingStepError = None,
+    ):
+        response = self.responses[self.case_index]["responses"][
+            str(ai_implementation_id)
+        ]
+        response["status"] = status.value
+        if error is not None:
+            response["error"] = error.value
+
+        report = {
+            "responses": self.responses,
+            "statistics": {
+                "currentCaseIndex": self.case_index,
+                "totalCaseCount": len(self.cases),
+            },
+        }
+        self.update_state(status="PROCESSING", meta=report)
+
+    def processing(self, ai_implementation_id: UUID):
+        self._update_case_status(
+            ai_implementation_id, BenchmarkingStepStatus.PROCESSING
+        )
+
+    def completed(self, ai_implementation_id: UUID, response: dict):
+        self._update_case_status(
+            ai_implementation_id, BenchmarkingStepStatus.COMPLETED
+        )
+        self.responses[self.case_index]["responses"][
+            str(ai_implementation_id)
+        ]["value"] = response
+
+    def error(self, ai_implementation_id: UUID, error: BenchmarkingStepError):
+        self._update_case_status(
+            ai_implementation_id, BenchmarkingStepStatus.ERRORED, error
+        )
+
+
+@shared_task(bind=True)
+def run_benchmark(self, benchmarking_session_id):
+    # todo: make configurable in benchmarking session
+    TIMEOUT = 10
```

Comment:
Maybe instead of configuring it on the benchmarking session it could be configured in the settings (and even loaded from an environment variable).

Reply:
i guess we'll need to discuss this. i could imagine that for different test scenarios different values might be necessary. (e.g. development benchmarks with ~10 seconds, "real" benchmarks with only one second to prevent tampering/adjustments)
but the default value should probably be part of the settings. 
@maccinza can you maybe move the value to settings? you're probably much faster at it than i would be (with figuring out how the settings really work)

Reply:
Sure, I can do that one.
In my opinion, in the future, maybe it would be good to have a `BenchmarkingConfiguration` model where those kind of settings could be configured and when creating a `BenchmarkingSession` it should reference a `BenchmarkingConfiguration`.
This would make those configurations more dynamic and accessible to users.